third-party/cmake/BuildLuajit.cmake

```diff
@@ -47,9 +47,18 @@ if(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
 else()
   set(AMD64_ABI "")
 endif()
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  check_c_compiler_flag(-fno-stack-check HAS_NO_STACK_CHECK)
+  if(HAS_NO_STACK_CHECK)
+    set(NO_STACK_CHECK "CFLAGS+=-fno-stack-check")
+  endif()
+else()
+  set(NO_STACK_CHECK "")
+endif()
```

Comment:
My cmake-foo is bad. Do variables have to be defined before use or are they just empty in such cases (like in shell scripts)?

In the latter case, the `else()` could be left away. In the former case, `NO_STACK_CHECK` won't be defined if `Darwin` and empty `HAS_NO_STACK_CHECK`.

Reply:
I was mirroring the check before that, so I assume the answer is "yes". But let me check.

Reply:
Hmm, seems you were right, undefined variables are just empty. But since all the other checks set them explicitly, I'll add the missing case.

Reply:
I ended up coalescing the case distinction; making that check only for Darwin wasn't worth the nesting (with the possibility of missing a case...)

Reply:
I like it. Very readable, too.